bundles/auth/core/src/main/java/org/apache/sling/auth/core/impl/SlingAuthenticator.java

```diff
@@ -1639,4 +1639,35 @@ protected AbstractAuthenticationHandlerHolder createHolder(String path,
         }
     }
 
+    /**
+     * SLING-6053
+     * Auth handler is required for the node and for all his children.
+     * Solved the case when sibling node name startsWith same name.
+     * <p>
+     * Eg: node: /page & node /page1
+     */
+    static class AuthenticationHandlerPath {
+
+        private final String requestPath;
+        private final String handlerPath;
+
+        AuthenticationHandlerPath(final String requestPath, final String handlerPath) {
+            this.requestPath = requestPath;
+            this.handlerPath = handlerPath;
+        }
+
+        boolean isNodeRequiresAuthHandler() {
+            if (requestPath.equalsIgnoreCase(handlerPath)) {
+                return true;
+            }
+            if (requestPath.startsWith(handlerPath)) {
+                final String suffix = requestPath.substring(handlerPath.length());
+                //if is child node auth handler is applied;
+                return suffix.contains("/");
```

Comment:
what about the fact that we have `requestPath` = `page1\somesubepage` and `handlerPath`= `page`?